src/main/java/core/basesyntax/FigureSupplier.java

```diff
@@ -0,0 +1,126 @@
+            fourthLeg = randomNumberForLegs() + secondLeg;
+        }
+
+        legs[0] = firstLeg;
+        legs[1] = secondLeg;
+        legs[2] = firstLeg;
+        legs[3] = fourthLeg;
+        return legs;
+    }
+
+    private int[] rectangle() {
+        int[] legs = new int[4];
+        int firstLeg = randomNumberForLegs();
+        int secondLeg = randomNumberForLegs();
+
+        if (firstLeg != secondLeg && secondLeg > firstLeg) {
+            legs[0] = firstLeg;
+            legs[1] = secondLeg;
+            legs[2] = firstLeg;
+            legs[3] = secondLeg;
+        } else {
+            while (firstLeg != secondLeg && secondLeg > firstLeg) {
+                firstLeg = randomNumberForLegs();
+                secondLeg = randomNumberForLegs();
+            }
+            legs[0] = firstLeg;
+            legs[1] = secondLeg;
+            legs[2] = firstLeg;
+            legs[3] = secondLeg;
+        }
+        return legs;
+    }
+
+    private int[] rightTriangle() {
+        int[] legs = new int[3];
+        int firstLeg = randomNumberForLegs();
+        int thirdLeg = randomNumberForLegs();
+        int secondLeg = (int) Math.sqrt(firstLeg * firstLeg + thirdLeg * thirdLeg);
+        while (firstLeg == thirdLeg || firstLeg >= secondLeg
+                || firstLeg == 0 || thirdLeg == 0 || thirdLeg >= secondLeg) {
+            firstLeg = randomNumberForLegs();
+            thirdLeg = randomNumberForLegs();
+            secondLeg = (int) Math.sqrt(firstLeg * firstLeg + thirdLeg * thirdLeg);
+        }
+        legs[0] = firstLeg;
+        legs[1] = secondLeg;
+        legs[2] = thirdLeg;
+        return legs;
+    }
+
+    private int[] square() {
+        int[] legs = new int[1];
+        int firstLeg = randomNumberForLegs();
+        legs[0] = firstLeg;
+        return legs;
+    }
+
+    public Figure getRandomFigure() {
+        Figure randomFigure;
+        randomInt = random.nextInt(5) + 1;
+        switch (randomInt) {
+            case 1: {
+                randomFigure = new Circle(randomNumberForLegs());
+                randomFigure.draw();
+                break;
+            }
+            case 2: {
+                randomFigure = new IsoscelesTrapezoid(isoscelesTrapezoid());
+                randomFigure.draw();
+                break;
+            }
+            case 3: {
+                randomFigure = new Rectangle(rectangle());
+                randomFigure.draw();
+                break;
+            }
+            case 4: {
+                randomFigure = new RightTriangle(rightTriangle());
+                randomFigure.draw();
+                break;
+            }
+            case 5: {
+                randomFigure = new Square(square());
+                randomFigure.draw();
+                break;
+            }
+            default:
+                randomFigure = new Circle(10);
+                randomFigure.setColor("white");
+                randomFigure.draw();
+                break;

```

Comment:
Redundant as never used, we have separate method for default figure, you may use default section for last random figure

Reply:
++